docs/zed/README.md

```diff
@@ -1,98 +1,288 @@
+
+A pool is created with the [create command](#23-create)
+and a branch of a pool is created with the [branch command](#22-branch).
+
+A pool name can be any valid UTF-8 string and is allocated a unique ID
+when created.  The pool can be referred to by its name or by its ID.
+A pool may be renamed but the unique ID is always fixed.
+
+### 1.4.1 Commit Objects
+
+Data is added into a pool in atomic units called _commit objects_.
+
+Each commit object is assigned a global ID.
+Similar to Git, Zed commit objects are arranged into a tree and
+represent the entire commit history of the lake.
+
+> Technically speaking, Git can merge from multiple parents and thus
+Git commits form a directed acyclic graph instead of a tree;
+Zed does not currently support multiple parents in the commit object history.
+
+A branch is simply a named pointer to a commit object in the Zed lake
+and like a pool, a branch name can be any valid UTF-8 string.
+Consistent updates to a branch are made by writing a new commit object that
+points to the previous tip of branch and updating the branch to point at
+the new commit object.  This update may be made with a transaction constraint
+(e.g., requiring that the previous branch tip is the same as the
+commit object's parent); if the constraint is violated, then the transaction
+is aborted.
+
+The _working branch_ of a pool may be selected on any command with the `-use` option
+or may be persisted across commands with the [use command](#214-use) so that
+`-use` does not have to be specified on each command-line.  For interactive
+workflows, the `use` command is convenient but for automated workflows
+in scripts, it is good practice to explicitly specify the branch in each
+command invocation with `-use` option.
+
+### 1.4.2 Commitish
+
+Many `zed` commands operate with respect to a commit object.
+While commit objects are always referenceable by their commit ID, it is also convenient
+to refer to the commit object at the tip of a branch.
+
+The entity that represents either a commit ID or a branch is called a _commitish_.
+A commitish is always relative to the pool and has the form:
+* `<pool>@<id>` or
+* `<pool>@<branch>`
+
+where `<pool>` is a pool name or pool ID, `<id>` is a commit object ID, and
+and `<branch>` is a branch name.
+
+In particular, the working branch set by the [use command](#214-use) is a commitish.
+
+A commitish may be abbreviated in several ways where the missing detail is
+obtained from the working-branch commitish, e.g.,
+* `<pool>` - When just a pool name is given, then the comittish is assumed to be
+`<pool>@main`.
+* `@<id>` or `<id>`- When an ID is given (optionally with the `@` prefix), then the commitish is assumed to be `<pool>@<id>` where `<pool>` is obtained from the working-branch commitish.
+* `@<branch>` - When a branch name is given with the `@` prefix, then the commitish is assumed to be `<pool>@<id>` where `<pool>` is obtained from the working-branch commitish.
```

Comment:
I'm not sure what you mean.  We explain IDs at line 126 and it would be wordy to explain this again everywhere they are referenced.  Or maybe you have a different point?

Reply:
@mccanne: Regarding the line in question:

> \* \`@\<branch>\` - When a branch name is given with the \`@\` prefix, then the commitish is assumed to be \`\<pool>@\<id>\` where \`\<pool>\` is obtained from the working-branch commitish.

It says how it's "assumed to be `<pool>@<id>`" but in my mind that opens the question of "Which ID?" I'm assuming it's the tip of the referenced branch, but I also figured it wouldn't hurt to be explicit. It stood out to me because in both of the prior two bullets, the origin of each thing referenced in `<>`  is explained.